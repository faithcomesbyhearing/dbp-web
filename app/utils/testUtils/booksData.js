export const bookData = {
  OT: [
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
        25,
        26,
        27,
        28,
        29,
        30,
        31,
        32,
        33,
        34,
        35,
        36,
        37,
        38,
        39,
        40,
        41,
        42,
        43,
        44,
        45,
        46,
        47,
        48,
        49,
        50,
      ],
      book_group: 'The Law',
      name: 'Genesis',
      testament: 'OT',
      book_id: 'GEN',
      book_order: 1,
      testament_order: 1,
      book_id_osis: 'Gen',
      book_id_usfx: 'GN',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
        25,
        26,
        27,
        28,
        29,
        30,
        31,
        32,
        33,
        34,
        35,
        36,
        37,
        38,
        39,
        40,
      ],
      book_group: 'The Law',
      name: 'Exodus',
      testament: 'OT',
      book_id: 'EXO',
      book_order: 2,
      testament_order: 2,
      book_id_osis: 'Exod',
      book_id_usfx: 'EX',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
        25,
        26,
        27,
      ],
      book_group: 'The Law',
      name: 'Leviticus',
      testament: 'OT',
      book_id: 'LEV',
      book_order: 3,
      testament_order: 3,
      book_id_osis: 'Lev',
      book_id_usfx: 'LV',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
        25,
        26,
        27,
        28,
        29,
        30,
        31,
        32,
        33,
        34,
        35,
        36,
      ],
      book_group: 'The Law',
      name: 'Numbers',
      testament: 'OT',
      book_id: 'NUM',
      book_order: 4,
      testament_order: 4,
      book_id_osis: 'Num',
      book_id_usfx: 'NU',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
        25,
        26,
        27,
        28,
        29,
        30,
        31,
        32,
        33,
        34,
      ],
      book_group: 'The Law',
      name: 'Deuteronomy',
      testament: 'OT',
      book_id: 'DEU',
      book_order: 5,
      testament_order: 5,
      book_id_osis: 'Deut',
      book_id_usfx: 'DT',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
      ],
      book_group: 'Historical Books',
      name: 'Joshua',
      testament: 'OT',
      book_id: 'JOS',
      book_order: 6,
      testament_order: 6,
      book_id_osis: 'Josh',
      book_id_usfx: 'JS',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
      ],
      book_group: 'Historical Books',
      name: 'Judges',
      testament: 'OT',
      book_id: 'JDG',
      book_order: 7,
      testament_order: 7,
      book_id_osis: 'Judg',
      book_id_usfx: 'JG',
    },
    {
      chapters: [1, 2, 3, 4],
      book_group: 'Historical Books',
      name: 'Ruth',
      testament: 'OT',
      book_id: 'RUT',
      book_order: 8,
      testament_order: 8,
      book_id_osis: 'Ruth',
      book_id_usfx: 'RT',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
        25,
        26,
        27,
        28,
        29,
        30,
        31,
      ],
      book_group: 'Historical Books',
      name: '1 Samuel',
      testament: 'OT',
      book_id: '1SA',
      book_order: 9,
      testament_order: 9,
      book_id_osis: '1Sam',
      book_id_usfx: 'S1',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
      ],
      book_group: 'Historical Books',
      name: '2 Samuel',
      testament: 'OT',
      book_id: '2SA',
      book_order: 10,
      testament_order: 10,
      book_id_osis: '2Sam',
      book_id_usfx: 'S2',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
      ],
      book_group: 'Historical Books',
      name: '1 Kings',
      testament: 'OT',
      book_id: '1KI',
      book_order: 11,
      testament_order: 11,
      book_id_osis: '1Kgs',
      book_id_usfx: 'K1',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
        25,
      ],
      book_group: 'Historical Books',
      name: '2 Kings',
      testament: 'OT',
      book_id: '2KI',
      book_order: 12,
      testament_order: 12,
      book_id_osis: '2Kgs',
      book_id_usfx: 'K2',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
        25,
        26,
        27,
        28,
        29,
      ],
      book_group: 'Historical Books',
      name: '1 Chronicles',
      testament: 'OT',
      book_id: '1CH',
      book_order: 13,
      testament_order: 13,
      book_id_osis: '1Chr',
      book_id_usfx: 'R1',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
        25,
        26,
        27,
        28,
        29,
        30,
        31,
        32,
        33,
        34,
        35,
        36,
      ],
      book_group: 'Historical Books',
      name: '2 Chronicles',
      testament: 'OT',
      book_id: '2CH',
      book_order: 14,
      testament_order: 14,
      book_id_osis: '2Chr',
      book_id_usfx: 'R2',
    },
    {
      chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
      book_group: 'Historical Books',
      name: 'Ezra',
      testament: 'OT',
      book_id: 'EZR',
      book_order: 15,
      testament_order: 15,
      book_id_osis: 'Ezra',
      book_id_usfx: 'ER',
    },
    {
      chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13],
      book_group: 'Historical Books',
      name: 'Nehemiah',
      testament: 'OT',
      book_id: 'NEH',
      book_order: 16,
      testament_order: 16,
      book_id_osis: 'Neh',
      book_id_usfx: 'NH',
    },
    {
      chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
      book_group: 'Historical Books',
      name: 'Esther',
      testament: 'OT',
      book_id: 'EST',
      book_order: 17,
      testament_order: 17,
      book_id_osis: 'Esth',
      book_id_usfx: 'ET',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
        25,
        26,
        27,
        28,
        29,
        30,
        31,
        32,
        33,
        34,
        35,
        36,
        37,
        38,
        39,
        40,
        41,
        42,
      ],
      book_group: 'Wisdom Books',
      name: 'Job',
      testament: 'OT',
      book_id: 'JOB',
      book_order: 18,
      testament_order: 18,
      book_id_osis: 'Job',
      book_id_usfx: 'JB',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
        25,
        26,
        27,
        28,
        29,
        30,
        31,
        32,
        33,
        34,
        35,
        36,
        37,
        38,
        39,
        40,
        41,
        42,
        43,
        44,
        45,
        46,
        47,
        48,
        49,
        50,
        51,
        52,
        53,
        54,
        55,
        56,
        57,
        58,
        59,
        60,
        61,
        62,
        63,
        64,
        65,
        66,
        67,
        68,
        69,
        70,
        71,
        72,
        73,
        74,
        75,
        76,
        77,
        78,
        79,
        80,
        81,
        82,
        83,
        84,
        85,
        86,
        87,
        88,
        89,
        90,
        91,
        92,
        93,
        94,
        95,
        96,
        97,
        98,
        99,
        100,
        101,
        102,
        103,
        104,
        105,
        106,
        107,
        108,
        109,
        110,
        111,
        112,
        113,
        114,
        115,
        116,
        117,
        118,
        119,
        120,
        121,
        122,
        123,
        124,
        125,
        126,
        127,
        128,
        129,
        130,
        131,
        132,
        133,
        134,
        135,
        136,
        137,
        138,
        139,
        140,
        141,
        142,
        143,
        144,
        145,
        146,
        147,
        148,
        149,
        150,
      ],
      book_group: 'Wisdom Books',
      name: 'Psalms',
      testament: 'OT',
      book_id: 'PSA',
      book_order: 19,
      testament_order: 19,
      book_id_osis: 'Ps',
      book_id_usfx: 'PS',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
        25,
        26,
        27,
        28,
        29,
        30,
        31,
      ],
      book_group: 'Wisdom Books',
      name: 'Proverbs',
      testament: 'OT',
      book_id: 'PRO',
      book_order: 20,
      testament_order: 20,
      book_id_osis: 'Prov',
      book_id_usfx: 'PR',
    },
    {
      chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12],
      book_group: 'Wisdom Books',
      name: 'Ecclesiastes',
      testament: 'OT',
      book_id: 'ECC',
      book_order: 21,
      testament_order: 21,
      book_id_osis: 'Eccl',
      book_id_usfx: 'EC',
    },
    {
      chapters: [1, 2, 3, 4, 5, 6, 7, 8],
      book_group: 'Wisdom Books',
      name: 'Song of Solomon',
      testament: 'OT',
      book_id: 'SNG',
      book_order: 22,
      testament_order: 22,
      book_id_osis: 'Song',
      book_id_usfx: 'SS',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
        25,
        26,
        27,
        28,
        29,
        30,
        31,
        32,
        33,
        34,
        35,
        36,
        37,
        38,
        39,
        40,
        41,
        42,
        43,
        44,
        45,
        46,
        47,
        48,
        49,
        50,
        51,
        52,
        53,
        54,
        55,
        56,
        57,
        58,
        59,
        60,
        61,
        62,
        63,
        64,
        65,
        66,
      ],
      book_group: 'Major Prophets',
      name: 'Isaiah',
      testament: 'OT',
      book_id: 'ISA',
      book_order: 23,
      testament_order: 23,
      book_id_osis: 'Isa',
      book_id_usfx: 'IS',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
        25,
        26,
        27,
        28,
        29,
        30,
        31,
        32,
        33,
        34,
        35,
        36,
        37,
        38,
        39,
        40,
        41,
        42,
        43,
        44,
        45,
        46,
        47,
        48,
        49,
        50,
        51,
        52,
      ],
      book_group: 'Major Prophets',
      name: 'Jeremiah',
      testament: 'OT',
      book_id: 'JER',
      book_order: 24,
      testament_order: 24,
      book_id_osis: 'Jer',
      book_id_usfx: 'JR',
    },
    {
      chapters: [1, 2, 3, 4, 5],
      book_group: 'Major Prophets',
      name: 'Lamentations',
      testament: 'OT',
      book_id: 'LAM',
      book_order: 25,
      testament_order: 25,
      book_id_osis: 'Lam',
      book_id_usfx: 'LM',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
        25,
        26,
        27,
        28,
        29,
        30,
        31,
        32,
        33,
        34,
        35,
        36,
        37,
        38,
        39,
        40,
        41,
        42,
        43,
        44,
        45,
        46,
        47,
        48,
      ],
      book_group: 'Major Prophets',
      name: 'Ezekiel',
      testament: 'OT',
      book_id: 'EZK',
      book_order: 26,
      testament_order: 26,
      book_id_osis: 'Ezek',
      book_id_usfx: 'EK',
    },
    {
      chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12],
      book_group: 'Major Prophets',
      name: 'Daniel',
      testament: 'OT',
      book_id: 'DAN',
      book_order: 27,
      testament_order: 27,
      book_id_osis: 'Dan',
      book_id_usfx: 'DN',
    },
    {
      chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14],
      book_group: 'Minor Prophets',
      name: 'Hosea',
      testament: 'OT',
      book_id: 'HOS',
      book_order: 28,
      testament_order: 28,
      book_id_osis: 'Hos',
      book_id_usfx: 'HS',
    },
    {
      chapters: [1, 2, 3],
      book_group: 'Minor Prophets',
      name: 'Joel',
      testament: 'OT',
      book_id: 'JOL',
      book_order: 29,
      testament_order: 29,
      book_id_osis: 'Joel',
      book_id_usfx: 'JL',
    },
    {
      chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9],
      book_group: 'Minor Prophets',
      name: 'Amos',
      testament: 'OT',
      book_id: 'AMO',
      book_order: 30,
      testament_order: 30,
      book_id_osis: 'Amos',
      book_id_usfx: 'AM',
    },
    {
      chapters: [1],
      book_group: 'Minor Prophets',
      name: 'Obadiah',
      testament: 'OT',
      book_id: 'OBA',
      book_order: 31,
      testament_order: 31,
      book_id_osis: 'Obad',
      book_id_usfx: 'OB',
    },
    {
      chapters: [1, 2, 3, 4],
      book_group: 'Minor Prophets',
      name: 'Jonah',
      testament: 'OT',
      book_id: 'JON',
      book_order: 32,
      testament_order: 32,
      book_id_osis: 'Jonah',
      book_id_usfx: 'JH',
    },
    {
      chapters: [1, 2, 3, 4, 5, 6, 7],
      book_group: 'Minor Prophets',
      name: 'Micah',
      testament: 'OT',
      book_id: 'MIC',
      book_order: 33,
      testament_order: 33,
      book_id_osis: 'Mic',
      book_id_usfx: 'MC',
    },
    {
      chapters: [1, 2, 3],
      book_group: 'Minor Prophets',
      name: 'Nahum',
      testament: 'OT',
      book_id: 'NAM',
      book_order: 34,
      testament_order: 34,
      book_id_osis: 'Nah',
      book_id_usfx: 'NM',
    },
    {
      chapters: [1, 2, 3],
      book_group: 'Minor Prophets',
      name: 'Habakkuk',
      testament: 'OT',
      book_id: 'HAB',
      book_order: 35,
      testament_order: 35,
      book_id_osis: 'Hab',
      book_id_usfx: 'HK',
    },
    {
      chapters: [1, 2, 3],
      book_group: 'Minor Prophets',
      name: 'Zephaniah',
      testament: 'OT',
      book_id: 'ZEP',
      book_order: 36,
      testament_order: 36,
      book_id_osis: 'Zeph',
      book_id_usfx: 'ZP',
    },
    {
      chapters: [1, 2],
      book_group: 'Minor Prophets',
      name: 'Haggai',
      testament: 'OT',
      book_id: 'HAG',
      book_order: 37,
      testament_order: 37,
      book_id_osis: 'Hag',
      book_id_usfx: 'HG',
    },
    {
      chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14],
      book_group: 'Minor Prophets',
      name: 'Zechariah',
      testament: 'OT',
      book_id: 'ZEC',
      book_order: 38,
      testament_order: 38,
      book_id_osis: 'Zech',
      book_id_usfx: 'ZC',
    },
    {
      chapters: [1, 2, 3, 4],
      book_group: 'Minor Prophets',
      name: 'Malachi',
      testament: 'OT',
      book_id: 'MAL',
      book_order: 39,
      testament_order: 39,
      book_id_osis: 'Mal',
      book_id_usfx: 'ML',
    },
  ],
  NT: [
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
        25,
        26,
        27,
        28,
      ],
      book_group: 'Gospels',
      name: 'Matthew',
      testament: 'NT',
      book_id: 'MAT',
      book_order: 41,
      testament_order: 1,
      book_id_osis: 'Matt',
      book_id_usfx: 'MT',
    },
    {
      chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16],
      book_group: 'Gospels',
      name: 'Mark',
      testament: 'NT',
      book_id: 'MRK',
      book_order: 42,
      testament_order: 2,
      book_id_osis: 'Mark',
      book_id_usfx: 'MK',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
      ],
      book_group: 'Gospels',
      name: 'Luke',
      testament: 'NT',
      book_id: 'LUK',
      book_order: 43,
      testament_order: 3,
      book_id_osis: 'Luke',
      book_id_usfx: 'LK',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
      ],
      book_group: 'Gospels',
      name: 'John',
      testament: 'NT',
      book_id: 'JHN',
      book_order: 44,
      testament_order: 4,
      book_id_osis: 'John',
      book_id_usfx: 'JN',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
        25,
        26,
        27,
        28,
      ],
      book_group: 'Apostolic History',
      name: 'Acts',
      testament: 'NT',
      book_id: 'ACT',
      book_order: 45,
      testament_order: 5,
      book_id_osis: 'Acts',
      book_id_usfx: 'AC',
    },
    {
      chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16],
      book_group: 'Pauline Epistles',
      name: 'Romans',
      testament: 'NT',
      book_id: 'ROM',
      book_order: 46,
      testament_order: 6,
      book_id_osis: 'Rom',
      book_id_usfx: 'RM',
    },
    {
      chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16],
      book_group: 'Pauline Epistles',
      name: '1 Corinthians',
      testament: 'NT',
      book_id: '1CO',
      book_order: 47,
      testament_order: 7,
      book_id_osis: '1Cor',
      book_id_usfx: 'C1',
    },
    {
      chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13],
      book_group: 'Pauline Epistles',
      name: '2 Corinthians',
      testament: 'NT',
      book_id: '2CO',
      book_order: 48,
      testament_order: 8,
      book_id_osis: '2Cor',
      book_id_usfx: 'C2',
    },
    {
      chapters: [1, 2, 3, 4, 5, 6],
      book_group: 'Pauline Epistles',
      name: 'Galatians',
      testament: 'NT',
      book_id: 'GAL',
      book_order: 49,
      testament_order: 9,
      book_id_osis: 'Gal',
      book_id_usfx: 'GL',
    },
    {
      chapters: [1, 2, 3, 4, 5, 6],
      book_group: 'Pauline Epistles',
      name: 'Ephesians',
      testament: 'NT',
      book_id: 'EPH',
      book_order: 50,
      testament_order: 10,
      book_id_osis: 'Eph',
      book_id_usfx: 'EP',
    },
    {
      chapters: [1, 2, 3, 4],
      book_group: 'Pauline Epistles',
      name: 'Philippians',
      testament: 'NT',
      book_id: 'PHP',
      book_order: 51,
      testament_order: 11,
      book_id_osis: 'Phil',
      book_id_usfx: 'PP',
    },
    {
      chapters: [1, 2, 3, 4],
      book_group: 'Pauline Epistles',
      name: 'Colossians',
      testament: 'NT',
      book_id: 'COL',
      book_order: 52,
      testament_order: 12,
      book_id_osis: 'Col',
      book_id_usfx: 'CL',
    },
    {
      chapters: [1, 2, 3, 4, 5],
      book_group: 'Pauline Epistles',
      name: '1 Thessalonians',
      testament: 'NT',
      book_id: '1TH',
      book_order: 53,
      testament_order: 13,
      book_id_osis: '1Thess',
      book_id_usfx: 'H1',
    },
    {
      chapters: [1, 2, 3],
      book_group: 'Pauline Epistles',
      name: '2 Thessalonians',
      testament: 'NT',
      book_id: '2TH',
      book_order: 54,
      testament_order: 14,
      book_id_osis: '2Thess',
      book_id_usfx: 'H2',
    },
    {
      chapters: [1, 2, 3, 4, 5, 6],
      book_group: 'Pauline Epistles',
      name: '1 Timothy',
      testament: 'NT',
      book_id: '1TI',
      book_order: 55,
      testament_order: 15,
      book_id_osis: '1Tim',
      book_id_usfx: 'T1',
    },
    {
      chapters: [1, 2, 3, 4],
      book_group: 'Pauline Epistles',
      name: '2 Timothy',
      testament: 'NT',
      book_id: '2TI',
      book_order: 56,
      testament_order: 16,
      book_id_osis: '2Tim',
      book_id_usfx: 'T2',
    },
    {
      chapters: [1, 2, 3],
      book_group: 'Pauline Epistles',
      name: 'Titus',
      testament: 'NT',
      book_id: 'TIT',
      book_order: 57,
      testament_order: 17,
      book_id_osis: 'Titus',
      book_id_usfx: 'TT',
    },
    {
      chapters: [1],
      book_group: 'Pauline Epistles',
      name: 'Philemon',
      testament: 'NT',
      book_id: 'PHM',
      book_order: 58,
      testament_order: 18,
      book_id_osis: 'Phlm',
      book_id_usfx: 'PM',
    },
    {
      chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13],
      book_group: 'Pauline Epistles',
      name: 'Hebrews',
      testament: 'NT',
      book_id: 'HEB',
      book_order: 59,
      testament_order: 19,
      book_id_osis: 'Heb',
      book_id_usfx: 'HB',
    },
    {
      chapters: [1, 2, 3, 4, 5],
      book_group: 'General Epistles',
      name: 'James',
      testament: 'NT',
      book_id: 'JAS',
      book_order: 60,
      testament_order: 20,
      book_id_osis: 'Jas',
      book_id_usfx: 'JM',
    },
    {
      chapters: [1, 2, 3, 4, 5],
      book_group: 'General Epistles',
      name: '1 Peter',
      testament: 'NT',
      book_id: '1PE',
      book_order: 61,
      testament_order: 21,
      book_id_osis: '1Pet',
      book_id_usfx: 'P1',
    },
    {
      chapters: [1, 2, 3],
      book_group: 'General Epistles',
      name: '2 Peter',
      testament: 'NT',
      book_id: '2PE',
      book_order: 62,
      testament_order: 22,
      book_id_osis: '2Pet',
      book_id_usfx: 'P2',
    },
    {
      chapters: [1, 2, 3, 4, 5],
      book_group: 'General Epistles',
      name: '1 John',
      testament: 'NT',
      book_id: '1JN',
      book_order: 63,
      testament_order: 23,
      book_id_osis: '1John',
      book_id_usfx: 'J1',
    },
    {
      chapters: [1],
      book_group: 'General Epistles',
      name: '2 John',
      testament: 'NT',
      book_id: '2JN',
      book_order: 64,
      testament_order: 24,
      book_id_osis: '2John',
      book_id_usfx: 'J2',
    },
    {
      chapters: [1],
      book_group: 'General Epistles',
      name: '3 John',
      testament: 'NT',
      book_id: '3JN',
      book_order: 65,
      testament_order: 25,
      book_id_osis: '3John',
      book_id_usfx: 'J3',
    },
    {
      chapters: [1],
      book_group: 'General Epistles',
      name: 'Jude',
      testament: 'NT',
      book_id: 'JUD',
      book_order: 66,
      testament_order: 26,
      book_id_osis: 'Jude',
      book_id_usfx: 'JD',
    },
    {
      chapters: [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
      ],
      book_group: 'Apocalypse',
      name: 'Revelation',
      testament: 'NT',
      book_id: 'REV',
      book_order: 67,
      testament_order: 27,
      book_id_osis: 'Rev',
      book_id_usfx: 'RV',
    },
  ],
};

export const books = [
	{
		chapters: [1, 2, 3, 4],
		book_group: 'Minor Prophets',
		name: 'Malachi',
		testament: 'OT',
		book_id: 'MAL',
		book_order: 39,
		testament_order: 39,
		book_id_osis: 'Mal',
		book_id_usfx: 'ML',
	},
	{
		chapters: [
			1,
			2,
			3,
			4,
			5,
			6,
			7,
			8,
			9,
			10,
			11,
			12,
			13,
			14,
			15,
			16,
			17,
			18,
			19,
			20,
			21,
			22,
			23,
			24,
			25,
			26,
			27,
			28,
		],
		book_group: 'Gospels',
		name: 'Matthew',
		testament: 'NT',
		book_id: 'MAT',
		book_order: 41,
		testament_order: 1,
		book_id_osis: 'Matt',
		book_id_usfx: 'MT',
	},
	{
		chapters: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16],
		book_group: 'Gospels',
		name: 'Mark',
		testament: 'NT',
		book_id: 'MRK',
		book_order: 42,
		testament_order: 2,
		book_id_osis: 'Mark',
		book_id_usfx: 'MK',
	},
];
