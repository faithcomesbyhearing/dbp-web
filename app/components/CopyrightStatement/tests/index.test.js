import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'; // Provides matchers like `toBeInTheDocument`
import CopyrightStatement from '..';
import { copyrights } from '../../../utils/testUtils/copyrightData';

/* eslint-disable react/prop-types */
jest.mock('react-intl', () => ({
	FormattedMessage: ({ defaultMessage }) => <span>{defaultMessage}</span>,
	defineMessages: (messages) => messages,
}));
/* eslint-enable react/prop-types */

const props = {
	organizations: copyrights.newTestament.text.organizations,
	testament: 'new_testament',
	type: 'text',
};

describe('CopyrightStatement component', () => {
	it('should match snapshot with valid new testament text props', () => {
		const { container } = render(<CopyrightStatement {...props} />);
		expect(container).toMatchSnapshot();
	});

	it('should match snapshot with valid new testament audio props', () => {
		const { container } = render(
			<CopyrightStatement
				organizations={copyrights.newTestament.audio.organizations}
				testament="new_testament"
				type="audio"
			/>,
		);
		expect(container).toMatchSnapshot();
	});

	it('should render a logo if the organization has one', () => {
		// Render component
		render(<CopyrightStatement {...props} />);

		// Check that there is at least one image in the document
		const images = screen.queryAllByRole('img');
		expect(images.length).toBeGreaterThan(0);
	});

	it('should not render a logo if the organization does not have one', () => {
		const customProps = { ...props };
		delete customProps.organizations[0].logo;

		// Render component without logo
		render(<CopyrightStatement {...customProps} />);

		// Check that no images are rendered
		const images = screen.queryAllByRole('img');
		expect(images.length).toEqual(0);
	});
});
