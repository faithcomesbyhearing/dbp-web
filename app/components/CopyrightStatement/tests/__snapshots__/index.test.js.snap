// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyrightStatement component should match snapshot with valid new testament audio props 1`] = `
<div>
  <h3>
    <span>
      This Bible audio is brought to you by:
    </span>
  </h3>
  <a
    href="http://www.fcbh.org"
    rel="noopener"
    target="_blank"
  >
    <img
      alt="Copyright owners logo: Faith Comes By Hearing"
      class="image-landscape fcbh-copyright-logo"
      src="https://bible.cloud/images/logo/FCBH_Color.svg"
    />
  </a>
</div>
`;

exports[`CopyrightStatement component should match snapshot with valid new testament text props 1`] = `
<div>
  <h3>
    <span>
      This Bible text is brought to you by:
    </span>
  </h3>
  <a
    href="https://www.crossway.org"
    rel="noopener"
    target="_blank"
  >
    <img
      alt="Copyright owners logo: Crossway"
      class="image-landscape"
      src="https://images.bible.cloud/logo/crossway.svg"
    />
  </a>
  <h3>
    <span>
      This Bible text is brought to you by:
    </span>
  </h3>
  <a
    class="org-name"
    rel="noopener"
    target="_blank"
  >
    Good News Publishers
  </a>
</div>
`;
